### DISCUSSION TITLE
Does Milvus load non-vector fields and non-indexed scalar fields into memory?

### DISCUSSION THREAD

**Discussion by amir-hojjati** [milvus-io/milvus#24592] [Q&A and General discussion] [answered]
Hi,

I had a question regarding the new Milvus updates. Milvus 2.1 enables users to store string data types, so let's say we are storing unindexed string data in a varchar field addition to vectors and other indexed scalar fields. Does Milvus loads those string data into memory as well when calling collection.load()?

Thank you!

**Comment by yhmo** [comment, marked as answer]
Yes, varchar field and other scalars field data are loaded into memory when calling collection.load().
The reason is: after load(), user can call collection.query() to retrieve scalar data from memory quickly.

But original vectors data is not loaded into memory because vectors original data might be huge. If user called collection.query() to retrieve vector data, milvus read vectors from storage, not from memory. That is why the performance of query vectors data is poor.

**Reply by amir-hojjati** [reply]
Thanks, that makes sense. Is there any way not to load some fields to memory? Let's say I have an application that I have to keep the collection loaded in memory, and I just want to retrieve those string fields (which can be large chunks of text) from the results whenever I perform a collection.search().

**Reply by yhmo** [reply]
> Thanks, that makes sense. Is there any way not to load some fields to memory? Let's say I have an application that I have to keep the collection loaded in memory, and I just want to retrieve those string fields (which can be large chunks of text) from the results whenever I perform a collection.search().

Not supported, milvus manages data by segments, each segment contains many rows of all fields, we are not able to load a segment partly.

**Reply by xiaofan-luan** [reply]
> Thanks, that makes sense. Is there any way not to load some fields to memory? Let's say I have an application that I have to keep the collection loaded in memory, and I just want to retrieve those string fields (which can be large chunks of text) from the results whenever I perform a collection.search().

We have mmap features in 2.3 where you can only store all the scalar datas on local disk and swap in/out.
Milvus 2.3 formal release will be on mid june
